File: src/etch/bindings/csharp/msg/Validator_void.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Etch.Msg
{
    /// <summary>
    /// Validator for void
    /// </summary>
    public class Validator_void : Validator
    {
        /// <summary></summary>
        /// <param name="nDims">the number of dimensions. 0 for scalar</param>
        /// <returns>an instance of the validator</returns>
        public static Validator_void Get( int nDims )
        {
            if ( nDims != 0 )
                throw new ArgumentException( "nDims != 0" );

            if ( validator == null )
                validator = new Validator_void();

            return validator;
        }

        private static Validator_void validator;

        private Validator_void()
        {
            // nothing to do
        }

        public override Validator ElementValidator()
        {
            return null;
        }

        public override bool Validate( object value )
        {
            return value == null;
        }

        public override object ValidateValue(object value)
        {
            if (Validate(value))
                return value;

            throw new Exception("value not appropriate for void: " + value);
        }      
    }
}

File: src/etch/bindings/java/msg/Validator_void.java
package etch.bindings.java.msg;

import etch.bindings.java.transport.fmt.TypeCode;

/**
 * Validator for void.
 */
public class Validator_void extends Validator
{
	/**
	 * @param nDims the number of dimensions. 0 for a scalar.
	 * @return an instance of the validator.
	 */
	public static Validator_void get( int nDims )
    {
		if (nDims != 0)
			throw new IllegalArgumentException( "nDims != 0" );
		
		if (validator == null)
			validator = new Validator_void();
		
		return validator;
    }
	
	private static Validator_void validator;
	
	/**
	 * Constructs the Validator.
	 */
	private Validator_void()
	{
		// nothing to do.
	}

	@Override
    public Validator elementValidator()
    {
	    return null;
    }

	@Override
	final public boolean validate( Object value )
	{
		return value == null;
	}

	@Override
	public Byte checkValue( Object value )
	{
		return validate( value ) ? TypeCode.NULL : null;
	}
}
